hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/NFTOwnersChangeStep.java

```diff
@@ -89,11 +89,15 @@ public void doIn(final TransferContext transferContext) {
 
                 final var copyNft = nft.copyBuilder();
                 // If the nft owner is not set then set it to the treasury account
-                if (nft.hasOwnerId() && nft.ownerId().equals(AccountID.DEFAULT)) {
+                if (!(nft.hasOwnerId() && nft.ownerId().equals(AccountID.DEFAULT))) {
```

Comment:
I am not really familiar with the token service, but in my understanding the comment above does not match the code.

Reply:
The `DEFAULT` instances are special instances that are automatically generated. They have no field set. I believe we typically use these instances instead of `null`.

Maybe @mhess-swl has more insight.

Reply:
I've tested it out, and I think you're right. Changed it back.